### ISSUE TITLE
showStatusBar causes image tearing

### ISSUE THREAD

**Issue by namaenonaimumei** [nomacs/nomacs#497]
## Description
Setting showStatusBar to true causes tearing on image load
Not noticeable when fast-forwarding, happens regardless of image resolution
Tearing is the most noticeable vertically at middle of screen

## Reproducing
Issue seems to persist with stock settings adjusting `Image Lounge.conf` to
```
[AppSettings]
showStatusBar=true
```
Included video sample describing issue
(compositioning disabled since OBS wouldn't cooperate)

* [Expected](https://streamable.com/m8ud5r) (`showStatusBar=false`)

* [Current](https://streamable.com/hnkxfe) (`showStatusBar=true`, notice image squashing artifacts in centre)

## Tested Environment
Tested on nomacs-3.14 with Windows 7 SP1 x64 and archlinux/Plasma/KWin/Xorg11 OpenGL3.1
Both with nvidia proprietary drivers (multiscreen, tested with single aswell)

**Comment by diemmarkus** [comment]
can you try to disable anti aliasing (`View` > `Anti Aliasing`) and see if that changes anything?

**Comment by namaenonaimumei** [comment]
Setting `antiAliasing=false` seems to work as expected (no tearing)
I actually tried with anti-aliasing disabled in `Edit`>`Settings`>`View`>`Zoom` before reporting (that would be `highQualityAntiAliasing=false`), but to no difference

Setting like this
```
[DisplaySettings]
antiAliasing=false
highQualityAntiAliasing=true
``` 
also seems to not be causing any tearing, not sure whether disabling `antiAliasing` disables `highQualityAntiAliasing` aswell or are these two independent from each other

**Comment by namaenonaimumei** [comment]
@diemmarkus Not sure whether it was okay to test it as it is, but issue seems to still partially remain

This is a local test with d24f9219d7da30d770ae44fd13ff2f1e7d17553a applied - current latest master
(notice part at cursor position, resizing can be seen at image edges)
* [Version master/9903d11](https://streamable.com/vrv3ad) (`antiAliasing=true` `showStatusBar=true`)

As you can see tearing at middle seems to be fixed, but image resizing/squashing artifacts are now present instead at each image load (same behavior as with vertical middle line one, issue only happens when `showStatusBar=true`). This kind of resizing does not happen for [Current](https://streamable.com/hnkxfe)/[Expected](https://streamable.com/m8ud5r)


Tried with both `highQualityAntiAliasing` disabled/enabled just in case, but this doesn't seem to make any difference for this issue

**Comment by diemmarkus** [comment]
actually I don't know `highQualityAntiAliasing` (I guess it is deprecated if it was ever there). 

> As you can see tearing at middle seems to be fixed, but image resizing/squashing artifacts are now present instead at each image load

for sure. here is a little explanation how displaying works in nomacs (with anti-aliasing enabled):
- `raw`: load image and display it using Qt 
- `cached`: compute high-quality anti-aliased version of current zoom level
- when finished, replace the currently displayed image with the `cached`

this algorithm guarantees a) fast displaying and b) high-quality displaying. Now to the issue that is somewhat solved by now:
- the `raw` version can render rectangles that are not aligned with your display's pixel (say width=400.3 px)
- the `cached` version must be aligned in order to get the best (visual) result (-> width=400px)

that is what you noticed as squashing artifacts. While I would consider the tearing as a bug (we actually lost a row back then) - the squashing is related to the above mentioned algorithm. Would you still consider this as a bug?

btw. the relation to the statusbar is only that the width becomes fractional with the statusbar displayed and is a natural number else.

**Comment by namaenonaimumei** [comment]
I see now, thanks for taking the time and explaining what happened

Perhaps this could be mitigated in future by specifying/limiting global image display area for fill view to integer instead of filling entire space available (so referencing case in explanation - 400px vertical, so that transition from raw to cached view would be seamless, in that case 400->400px)

Either way, since this is expected behavior and bug has been fixed, will be closing this issue in 24 hours if that's fine

**Comment by diemmarkus** [comment]
thanks. You are right: I'll check if we can also get rid of the squashing (I just think it's not trival and/or might cause other issues...)

**Comment by namaenonaimumei** [comment]
@diemmarkus Sorry for noticing so late, checked out current latest master (couldn't confirm in latest stable, since crashes for arch, so had to rollback to 3.16 (issue already reported and fixed with de0250a1b717b55af94af8bf1ab0876bca7a4015)) and setting `antiAliasing=true` doesn't cause squashing anymore
Thank you for looking into it once again
